docs/design/RUC-on-type.md

```diff
@@ -0,0 +1,189 @@
+# RequiresUnreferencedCode on Types
+The RequiresUnreferencedCode being allowed on types its a feature that allows developers to quickly mark entire classes as dangerous for trimming. instead of having to mark every method in the class. This document will go over whats the behavior of certain scenarios when the RequiresUnreferencedCode attribute is placed on a type
+
+## RequiresUnreferencedCode on a type warns on instance constructor or static method access
+Having RequiresUnreferencedCode on the type implies that any instance constructor or static method is considered dangerous for trimming. Following are examples of this behavior:
+
+```C#
+[RequiresUnreferencedCode ("Message for --ClassWithRequiresUnreferencedCode--")]
+class ClassWithRequiresUnreferencedCode
+{
+    public static void StaticMethod () { }
+
+    public void NonStaticMethod () { }
+}
+
+static void TestRequiresUnreferencedCodeInClass ()
+{
+    // IL2026: TestRequiresUnreferencedCodeInClass(): Using method 'ClassWithRequiresUnreferencedCode.StaticMethod()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Message for --ClassWithRequiresUnreferencedCode--.
```

Comment:
More of a side note - we should probably tweak the message for IL2026 to include the option of having RUC on the type.